src/Raven.Server/Documents/Expiration/ExpiredDocumentsCleaner.cs

```diff
@@ -123,10 +124,30 @@ public DeleteExpiredDocumentsCommand(Dictionary<Slice, List<(Slice LowerId, Lazy
                 _database = database;
             }
 
-            public override int Execute(DocumentsOperationContext context)
+            protected override int ExecuteCmd(DocumentsOperationContext context)
             {
                 return DeletionCount = _database.DocumentsStorage.ExpirationStorage.DeleteExpiredDocuments(context, _expired);
             }
+
+            public override TransactionOperationsMerger.IReplayableCommandDto<TransactionOperationsMerger.MergedTransactionCommand> ToDto(JsonOperationContext context)
+            {
+                return new DeleteExpiredDocumentsCommandDto
+                {
+                    Expired = _expired.ToArray()
+                };
+            }
+        }
+    }
+
+    internal class DeleteExpiredDocumentsCommandDto : TransactionOperationsMerger.IReplayableCommandDto<ExpiredDocumentsCleaner.DeleteExpiredDocumentsCommand>
+    {
+        public ExpiredDocumentsCleaner.DeleteExpiredDocumentsCommand ToCommand(DocumentsOperationContext context, DocumentDatabase database)
+        {
+            Dictionary<Slice, List<(Slice LowerId, LazyStringValue Id)>> expired = Expired.ToDictionary(kv => kv.Key, kv => kv.Value);
```

Comment:
Why not clone it?

Reply:
https://www.newtonsoft.com/json/help/html/SerializationGuide.htm#Dictionarys

At the time I did not look into it in depth, but using the dictionary rose a problem
Now I've found that there is special regard to the key of the dictionary
that cause it not going through the SliceJsonConverter

Reply:
Use `new Dictionary<...>(Expires)`

`ToDictionary` is invoking two delegates per instance and can throw if there are duplicates (which isn't an issue when reading from a dictionary, but in that case, why bother with `ToDictionary` ?

Reply:
done